base/float.jl

```diff
@@ -88,18 +88,27 @@ exponent_mask(::Type{Float64}) =    0x7ff0_0000_0000_0000
 exponent_one(::Type{Float64}) =     0x3ff0_0000_0000_0000
 exponent_half(::Type{Float64}) =    0x3fe0_0000_0000_0000
 significand_mask(::Type{Float64}) = 0x000f_ffff_ffff_ffff
+mantissa_width(::Type{Float64}) = UInt64(52)
+exponent_width(::Type{Float64}) = UInt64(11)
 
 sign_mask(::Type{Float32}) =        0x8000_0000
 exponent_mask(::Type{Float32}) =    0x7f80_0000
 exponent_one(::Type{Float32}) =     0x3f80_0000
 exponent_half(::Type{Float32}) =    0x3f00_0000
 significand_mask(::Type{Float32}) = 0x007f_ffff
+mantissa_width(::Type{Float32}) = UInt32(23)
+exponent_width(::Type{Float32}) = UInt32(8)
 
 sign_mask(::Type{Float16}) =        0x8000
 exponent_mask(::Type{Float16}) =    0x7c00
 exponent_one(::Type{Float16}) =     0x3c00
 exponent_half(::Type{Float16}) =    0x3800
 significand_mask(::Type{Float16}) = 0x03ff
+mantissa_width(::Type{Float16}) = UInt16(10)
+exponent_width(::Type{Float16}) = UInt16(5)
+
+uintbits(x::T) where {T<:IEEEFloat} = reinterpret(Unsigned, x)
```

Comment:
The point I intended to make (but was terse and unclear, apologies for that) with my previous comment about this was that you can replace uses of `uintbits(x)` with `reinterpret(Unsigned, x)`, which is a pattern already in use elsewhere.